docs/guides/events/advanced.md

```diff
@@ -0,0 +1,115 @@
+# Advanced Usage
+
+```{attention}
+This guide only applies to GatewayBot. REST-only applications cannot receive events through the gateway.
+```
+
+## wait_for
+
+Sometimes you may want to wait for an event to be received in a procedural manner, then proceed
+with execution.
+
+With `wait_for()` you can block until you receive an event with a given predicate and timeout.
```

Comment:
```suggestion
With `wait_for()` you can block until you receive an event with a given predicate or the function times out.
```

Reply:
Disagree with changing the wording here honestly.

Reply:
The current wording sounds like the event has a timeout attribute that has to match a certain value, but that's not a thing lol.

Reply:
My suggestion (basically the same as Lunar's):
```suggestion
With `wait_for()` you can block until you receive an event with a given predicate, or until a timeout is reached.
```